javascript/javascript3/week2/promise-visual/main.js

```diff
@@ -0,0 +1,41 @@
+const redDomElement = document.querySelector('li:nth-child(1)');
+const blueDomElement = document.querySelector('li:nth-child(2)');
+const greenDomElement = document.querySelector('li:nth-child(3)');
+
+const redLocationTarget = {x: 20, y: 300};
+const blueLocationTarget = {x: 400, y: 300};
+const greenLocationTarget = {x: 400, y: 20};
+
+// translateOneByOne 
+
+function translateOneByOne(){
+    return new Promise(() => {
+        moveElement(redDomElement, redLocationTarget)
+            .then(()=> {
+                console.log('red dot has moved to its spot.')
+                return moveElement(blueDomElement, blueLocationTarget);
+            })
+            .then(() => {
+                console.log('blue dot has moved to its spot.')
+                return moveElement(greenDomElement, greenLocationTarget);
+            })
+            .then(()=> {console.log('green dot has moved to its spot.')});
+    })
+}
+    
+translateOneByOne()
+
+//translateAllAtOnce
+
+function translateAllAtOnce() {
+    const redPromise = moveElement(redDomElement, redLocationTarget);
+    const greenPromise = moveElement(blueDomElement, blueLocationTarget);
+    const bluePromise = moveElement(greenDomElement, greenLocationTarget);
+
+    Promise.all([redPromise, greenPromise, bluePromise])
```

Comment:
Good job on this task too 👍 These are is just minor suggestions.

Instead of creating these variables, you could the array directly, like:
```
const actions = [
  moveElement(redDomElement, redLocationTarget),
  moveElement(blueDomElement, blueLocationTarget),
  moveElement(greenDomElement, greenLocationTarget),
]

Promise.all(actions)
 .then(...
```

Reply:
Thank you for your review. I implemented changes accordingly :)